pkg/component/labels/labels.go

```diff
@@ -5,30 +5,46 @@ import (
 	"github.com/redhat-developer/odo/pkg/util"
 )
 
-// ComponentLabel is a label key used to identify the component name
-const ComponentLabel = "app.kubernetes.io/instance"
+// ComponentKubernetesInstanceLabel is a label key used to identify the component name
+const ComponentKubernetesInstanceLabel = "app.kubernetes.io/instance"
 
-// ComponentTypeLabel is Kubernetes label that identifies the type of a component being used
-const ComponentTypeLabel = "app.kubernetes.io/name"
+// ComponentManagedByLabel is the label Kubernetes used to determine what the component is managed by
+const ComponentManagedByLabel = "app.kubernetes.io/managed-by"
 
-const ComponentTypeAnnotation = "odo.dev/project-type"
+// ComponentKubernetesNameLabel is Kubernetes label that identifies the type of a component being used
+const ComponentKubernetesNameLabel = "app.kubernetes.io/name"
+
+// ComponentUnknownLabel is the label that is used to display something we do not know
+const ComponentUnknownLabel = "<unknown>"
+
+// ComponentDevLabel ...
+const ComponentDevLabel = "Dev"
 
 // ComponentDeployLabel ...
 const ComponentDeployLabel = "Deploy"
 
-// ComponentModeLabel ...
-const ComponentModeLabel = "odo.dev/mode"
+// ComponentNoneLabel ...
+const ComponentNoneLabel = "None"
+
+// ComponentPushedLabel ...
+const ComponentPushedLabel = "Pushed"
+
+// KubernetesManagedByLabel ...
+const KubernetesManagedByLabel = "app.kubernetes.io/managed-by"
 
 // ComponentProjectTypeLabel ...
 const ComponentProjectTypeLabel = "odo.dev/project-type"
```

Comment:
This should not be a label. 
Its value directly maps to devfiles `metadata.projectType` and that can be a regular string with spaces and special characters, which could make it an invalid value for label value.

Also there is really no reason to have this as label as we never use it as a label selector

Reply:
in the code this is actually used for annotation, it should be renamed